core/shared/src/main/scala/spire/algebra/package.scala

```diff
@@ -0,0 +1,162 @@
+package spire
+
+package object algebra {
+
+  type Eq[A] = _root_.algebra.Eq[A]
+  val Eq = _root_.algebra.Eq
+
+  type PartialOrder[A] = _root_.algebra.PartialOrder[A]
+  val PartialOrder = _root_.algebra.PartialOrder
+
+  type Order[A] = _root_.algebra.Order[A]
+  val Order = _root_.algebra.Order
+
+  type Semigroup[A] = _root_.algebra.Semigroup[A]
+  val Semigroup = _root_.algebra.Semigroup
+
+  type CSemigroup[A] = _root_.algebra.CommutativeSemigroup[A]
+  val CSemigroup = _root_.algebra.CommutativeSemigroup
+
+  type Monoid[A] = _root_.algebra.Monoid[A]
+  val Monoid = _root_.algebra.Monoid
+
+  type CMonoid[A] = _root_.algebra.CommutativeMonoid[A]
+  val CMonoid = _root_.algebra.CommutativeMonoid
+
+  type Group[A] = _root_.algebra.Group[A]
+  val Group = _root_.algebra.Group
+
+  type AbGroup[A] = _root_.algebra.CommutativeGroup[A]
+  val AbGroup = _root_.algebra.CommutativeGroup
+
+  type AdditiveSemigroup[A] = _root_.algebra.ring.AdditiveSemigroup[A]
+  val AdditiveSemigroup = _root_.algebra.ring.AdditiveSemigroup
+
+  type AdditiveCSemigroup[A] = _root_.algebra.ring.AdditiveCommutativeSemigroup[A]
+  val AdditiveCSemigroup = _root_.algebra.ring.AdditiveCommutativeSemigroup
+
+  type AdditiveMonoid[A] = _root_.algebra.ring.AdditiveMonoid[A]
+  val AdditiveMonoid = _root_.algebra.ring.AdditiveMonoid
+
+  type AdditiveCMonoid[A] = _root_.algebra.ring.AdditiveCommutativeMonoid[A]
+  val AdditiveCMonoid = _root_.algebra.ring.AdditiveCommutativeMonoid
+
+  type AdditiveGroup[A] = _root_.algebra.ring.AdditiveGroup[A]
+  val AdditiveGroup = _root_.algebra.ring.AdditiveGroup
+
+  type AdditiveAbGroup[A] = _root_.algebra.ring.AdditiveCommutativeGroup[A]
+  val AdditiveAbGroup = _root_.algebra.ring.AdditiveCommutativeGroup
+
+  object Additive {
```

Comment:
I don't have strong opinions either way, but if you wanted to clean this up, `Additive` and `Multiplicative` can stay in its own file.